app/views/shared/require_login.html.erb

```diff
@@ -0,0 +1,4 @@
+<h2>
+  You must <%= link_to "register", register_path %> or 
+  <%= link_to "log in", login_path %> to access this.
+</h2>
```

Comment:
This is rendered if a visitor manages to try to access the checkout pages even when they don't have a link.

Reply:
This is a good edge case to catch